src/enum.cr

```diff
@@ -302,6 +302,20 @@ struct Enum
     {% end %}
   end
 
+  # Returns the number of enum members.
+  #
+  # ```
+  # Color.size  # => 3
+  # IOMode.size # => 3
+  # ```
+  def self.size : Int32
+    {% if @type.has_attribute?("Flags") %}
+      {{ @type.constants.size - 2 }}
```

Comment:
Actually this should be the popcount of all the constants `&`ed together.

Reply:
The what? what does `popcount` mean?

Reply:
Number of non-zero bits. That's significant if there is a flag that sets multiple bits.

I'm not sure if that's the right behaviour, but generally this method is probably not that much relevant for Flag enums anyway.

Reply:
For enums it should be the number of *unique* values the enum has (currently this is not unique), for flags it should be the number of actual flag bits covered by all the options. For example in some enums you have `A = 0b01`, `B = 0b10` and `Both = 0b11`, That's 3 flag constants but only 2 flag bits, which `popcount` solves.

Reply:
The question is, should `.size` return the number of constants or the number of unique values?